cppcache/src/TcrMessage.hpp

```diff
@@ -1178,194 +910,73 @@ class TcrMessageExecuteFunction : public TcrMessage {
-      return ChunkObjectType::NULL_OBJECT;
-    }
-
-    // TODO enum - wtf?
-    if (expectedFirstType > DSCode::FixedIDDefault) {
-      if (partType != expectedFirstType) {
-        char exMsg[256];
-        std::snprintf(exMsg, sizeof(exMsg),
-                      "TcrMessageHelper::readChunkPartHeader: "
-                      "%s: got unhandled object class = %" PRId8,
-                      methodName, static_cast<int8_t>(partType));
-        throw MessageException(exMsg);
-      }
-      // This is for GETALL
-      if (expectedFirstType == DSCode::FixedIDShort) {
-        compId = input.readInt16();
-      }  // This is for QUERY or REGISTER INTEREST.
-      else if (expectedFirstType == DSCode::FixedIDByte) {
-        compId = input.read();
-      }
-    }
-    if (compId != expectedPartType) {
-      char exMsg[256];
-      std::snprintf(
-          exMsg, sizeof(exMsg),
-          "TcrMessageHelper::readChunkPartHeader: "
-          "%s: got unhandled object type = %d, expected = %d, raw = %d",
-          methodName, compId, expectedPartType, rawByte);
-      throw MessageException(exMsg);
-    }
-    return ChunkObjectType::OBJECT;
-  }
-
-  inline static ChunkObjectType readChunkPartHeader(TcrMessage& msg,
-                                                    DataInput& input,
-                                                    const char* methodName,
-                                                    uint32_t& partLen,
-                                                    uint8_t isLastChunk) {
-    partLen = input.readInt32();
-    const auto isObj = input.readBoolean();
-
-    if (partLen == 0) {
-      // special null object is case for scalar query result
-      return ChunkObjectType::NULL_OBJECT;
-    } else if (!isObj) {
-      // otherwise we're currently always expecting an object
-      char exMsg[256];
-      std::snprintf(exMsg, 255,
-                    "TcrMessageHelper::readChunkPartHeader: "
-                    "%s: part is not object",
-                    methodName);
-      throw MessageException(exMsg);
-    }
-
-    const auto partType = static_cast<const DSCode>(input.read());
-    //  ugly hack to check for exception chunk
-    if (partType == DSCode::JavaSerializable) {
-      input.reset();
-      if (TcrMessageHelper::readExceptionPart(msg, input, isLastChunk)) {
-        msg.setMessageType(TcrMessage::EXCEPTION);
-        return ChunkObjectType::EXCEPTION;
-      } else {
-        char exMsg[256];
-        std::snprintf(
-            exMsg, 255,
-            "TcrMessageHelper::readChunkPartHeader: %s: cannot handle "
-            "java serializable object from server",
-            methodName);
-        throw MessageException(exMsg);
-      }
-    } else if (partType == DSCode::NullObj) {
-      // special null object is case for scalar query result
-      return ChunkObjectType::NULL_OBJECT;
-    }
-    return ChunkObjectType::OBJECT;
-  }
+  TcrMessageHelper() = delete;
```

Comment:
Deleted constructors and other special functions should always be `public`.

Reply:
Sure. Is there a reason for that?

Reply:
Assumption is you'll generally get better error messages, specifically in this case GCC version < 8.